src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useState, useCallback } from 'react';
+import { RouteComponentProps } from '@reach/router';
+import { formatISO, format } from 'date-fns';
+
+import Alert from 'src/components/Alert';
+import Button from 'src/components/inputs/Button';
+import DateField from 'src/components/fields/DateField';
+import SelectField from 'src/components/fields/SelectField';
+import TextAreaField from 'src/components/fields/TextAreaField';
+import TextField from 'src/components/fields/TextField';
+import TimeField from 'src/components/fields/TimeField';
+import { EventFormWrapper, Form } from './styled';
+import { EVENT_OPTIONS } from './constants';
+
+export type EventDataInput = {
+  id?: string;
+  name?: string;
+  description?: string;
+  startTime?: Date;
+  endTime?: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType?: string;
+};
+
+export type EventDataOutput = {
+  id?: string;
+  name: string;
+  description: string;
+  startTime: Date;
+  endTime: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType: string;
+};
+
+type EventFormProps = {
+  headerTitle: string;
+  onSubmit: (data: EventDataOutput) => Promise<void>;
+} & EventDataInput;
+
+function EventForm({ onSubmit, ...rest }: EventFormProps) {
```

Comment:
Personally, I prefer to destructure all of the variables instead of packing them into a `rest` variable whenever possible. My only reason for this though is that I think it looks cleaner without that `rest.*` prefix later on 😛.